### PR TITLE
test: integration tests for users

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

test: integration tests for users

## Testing

None.

### CODE REVIEW

File: packages/integration-tests/tests/users.test.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      },
+    })
+    .json<User>();
+};
+
+describe('admin console user management', () => {
+  it('should create user successfully', async () => {
+    const user = await createUser();
+
+    expect(user).toBeTruthy();
```
Comment (simeng-li) [inline packages/integration-tests/tests/users.test.ts:24]: Chould verify the username of the created user?

File: packages/integration-tests/tests/users.test.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+    expect(users.length).not.toBeLessThan(1);
+  });
+
+  it('should get user details successfully', async () => {
+    const user = await createUser();
+
+    const userDetails = await authedAdminApi.get(`users/${user.id}`).json<User>();
+
+    expect(userDetails).toBeTruthy();
```
Comment (simeng-li) [inline packages/integration-tests/tests/users.test.ts:39]: at least verify some attributes?  {} is truthy

File: packages/integration-tests/tests/users.test.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+      roleNames: ['admin'],
+    };
+
+    const updatedUser = await authedAdminApi
+      .patch(`users/${user.id}`, {
+        json: newUserData,
+      })
+      .json<User>();
+
+    expect(updatedUser).toMatchObject(newUserData);
```
Comment (simeng-li) [inline packages/integration-tests/tests/users.test.ts:60]: To contain? Response User type not guaranteed same as your Partial<User>

**Review by simeng-li — APPROVED** [review]

File: packages/integration-tests/tests/users.test.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    const user = await createUser();
+
+    expect(user).toBeTruthy();
+  });
+
+  it('should get user list successfully', async () => {
+    await createUser();
+    const users = await authedAdminApi.get('users').json<User[]>();
+
+    expect(users.length).not.toBeLessThan(1);
```
[the remark this answers is not in the document]
Comment (xiaoyijun) [reply packages/integration-tests/tests/users.test.ts:31]: Not here? This is a test to get the list and will verify the user in the creation test.

File: packages/integration-tests/tests/users.test.ts (hunk 3 above)
Comment (xiaoyijun) [reply packages/integration-tests/tests/users.test.ts:56]: `toMatchObject` is ok here?
image
image

Comment (simeng-li) [reply packages/integration-tests/tests/users.test.ts:56]: o ok my bad sounds good.

File: packages/integration-tests/tests/users.test.ts [hunk 5]
```diff
@@ -0,0 +1,88 @@
+import { User } from '@logto/schemas';
+
+import { authedAdminApi } from '@/api';
+import { generatePassword, generateUsername } from '@/utils';
+
+const createUser = async (username?: string) => {
+  const useUsername = username ?? generateUsername();
```
Comment (gao-sun) [inline packages/integration-tests/tests/users.test.ts:7]: 1. This name is confusing
2. I didn't see the usage for assigned username...

**Review by gao-sun — CHANGES_REQUESTED** [review]

File: packages/integration-tests/tests/users.test.ts (hunk 5 above)
Comment (xiaoyijun) [reply packages/integration-tests/tests/users.test.ts:7]: This is used in:

https://github.com/logto-io/logto/pull/1640/files#diff-2a55e5a4ae83380e5f03423dcb061b67bc3cc4a50be610327524085f51dd5864R21-R28

**Review by darcyYe — APPROVED** [review]

File: packages/integration-tests/tests/users.test.ts (hunk 5 above)
Comment (gao-sun) [reply packages/integration-tests/tests/users.test.ts:7]: per our discussion, instead of testing username, you should test user id.

**Review by gao-sun — APPROVED** [review]

**Pull request merged**